rts/motoko-rts/src/visitor/enhanced.rs

```diff
@@ -42,6 +42,12 @@ pub unsafe fn visit_pointer_fields<C, F, G>(
             }
         }
 
+        TAG_WEAK_REF => {
+            // WeakRef objects themselves are visited (so they don't get collected)
+            // but we deliberately do NOT visit the 'field' to avoid marking
+            // the weakly referenced object as live
```

Comment:
Visiting the pointer field is important to update its target (when alive) to new relocated location during GC update phase.

Reply:
In marking, it needs be skipped though

Reply:
I think it should be fixed now, please check.